opencog/nlp/chatbot/README

```diff
@@ -164,184 +111,112 @@ is as follows:
+   ''check_query_type''.This function will identify the speech act type of 
+   the sentence then the next appropriate processing functions will
+   be called. For instance currently  ''(wh_query_process query)'' will be 
+   invoked if the speech act tag is 'InterrogativeSpeechAct'  
+
+3) Question-Answering. Currently The bot can recognize speech act type of 
+   the utterance. specifically it identifies 'InterrogativeSpeechAct' ,
+   'TruthQuerySpeechAct and 'DeclarativeSpeechAct'
+
+   The bot can answer wh-questions ('InterrogativeSpeechAct') by using the 
+   fuzzy-pattern-matcher. Therefore we need to put some sentence related to 
+   the question in the atomspace before we ask the question.
+
+4) Relex2Logic replaces the old Semantic triples code. The current 
+   fuzzy-pattern-matcher uses the R2L output. Also PLN will use r2l output
+   for reasoning.
+
+5) Sureal and Microplanning is  used for Natural Language Generation.
+
+6) OpenPsi(not hookup)- will be used to make the chatbot to generate Goal
+   plus context oriented utterance. Here,different speech act schema will be
+   coded and those speech act will be triggered based on the Goal and context  
+
+7) Reasoning (not hookup). PLN,the core reasoning engine will be used.
+
+8) Anaphora resolution(not hookup) will be used  to resolve the  pronouns 
+
+9) Word-sense disambiguation (not hookup)
+   http://wiki.opencog.org/w/Word_sense_disambiguation
+
 
 Examples:
 ---------
-The following dialog works, or has worked, at various times in the past.
-Because the infrastructure is in development, any given snapshot may be
-broken.
-
-<me>         Fred threw a red ball
-<me>         John threw a green ball.
-<me>         Mary threw a blue rock
-<me>         who threw a ball?
-<cogita-bot> SVO pattern match found: John Fred
-<me>         who threw a red ball?
-<cogita-bot> SVO pattern match found: Fred
-<me>         What did Fred throw?
-<cogita-bot> SVO pattern match found: red ball
-
-<me>         Did Fred throw a ball?
-<cogita-bot> Truth query determined: Yes, verb was: throw
-
-<me>         Did Fred throw a red ball?
-<cogita-bot> Truth query determined: Yes, verb was: throw
-<me>         Did Fred throw a green ball?
-<cogita-bot> Truth query determined: No, not that I know of.
-
-<me>         The color of the book is red.
-<me>         What is the color of the book?
-<cogita-bot> Triples abstraction found: red
-
-<me>         the cat sat on the mat
-<me>         what did the cat sit on?
-<cogita-bot> Triples abstraction found: mat
-
-<me>         John and Berd turned into monsters
-<me>         who turned into monsters?
-<cogita-bot> Triples abstraction found: John Berd Berd John
-
-<me>         John made fun of Sarah.
-<me>         Who made fun of Sarah?
-<cogita-bot> SVO pattern match found: John
-
-Some things that don't work:
-"Did Fred throw a round ball?" -- doesn't know that balls are round.
-
-Untested, probably won't work:
```

Comment:
Could you restore the section below and above, marked as "examples that don't yet work, but should"?  We need to keep a list of statements and questions that are known to work, somewhere: in the README, for now, maybe in some other file later on, maybe as a part of a unit test.

Some of the questions above are fairly sophisticated, but still do not need PLN or inference in order to be answered.